account/watcher/controller_test.go

```diff
@@ -0,0 +1,363 @@
+package watcher
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	gomock "github.com/golang/mock/gomock"
+	"github.com/lightninglabs/pool/internal/test"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	errCtrlExpected = errors.New("random error")
+)
+
+var controllerLifeCycleTestCases = []struct {
+	name       string
+	mockSetter func(
+		mockChainNotifier *test.MockChainNotifierClient,
```

Comment:
nit: function definition can fit on one line to increase readability. Same below in the implementations.